### ISSUE TITLE
KeyError: u'app_function' on Django management commands

### ISSUE THREAD

**Issue by noelpuru** [Miserlou/Zappa#832]
```
Traceback (most recent call last):                                                                      
  File "/env/lib/site-packages/zappa/cli.py", line 2352, in handle           
    sys.exit(cli.handle())                                                                              
  File "/env/lib/site-packages/zappa/cli.py", line 456, in handle            
    self.dispatch_command(self.command, environment)                                                    
  File "/env/lib/site-packages/zappa/cli.py", line 476, in dispatch_command  
    if self.vargs['app_function'] is not None:                                                          
KeyError: u'app_function'                                                                             
```

I copied this command from the documentation:
`zappa manage staging "shell --version"`
but I got this error. This is after the initial deployment.

Environment
* Zappa version used: zappa==0.41.2
* Operating System and Python version: Python 2.7 on Windows 10
* Your `zappa_settings.py`: 
```
{
    "staging": {
        "aws_region": "ap-southeast-2", 
        "django_settings": "projname.settings_prod", 
        "profile_name": "default", 
        "s3_bucket": "projname"
    }
}
```

**Comment by Miserlou** [comment]
Good bug. Wonder if this is Windows related or just a fuckup.

**Comment by noelpuru** [comment]
I also tried it on Mac, same. Tried Django 1.11 too (currently using 1.8). My workaround is to downgrade to 0.41.1.

**Comment by majackson** [comment]
I was also having the above problem, and testing with the fix applied, I now get the error:

```
Traceback (most recent call last):
  File "/root/.venvs/shinobivenv/lib/python3.6/site-packages/zappa/cli.py", line 2353, in handle
    sys.exit(cli.handle())
  File "/root/.venvs/shinobivenv/lib/python3.6/site-packages/zappa/cli.py", line 456, in handle
    self.dispatch_command(self.command, environment)
  File "/root/.venvs/shinobivenv/lib/python3.6/site-packages/zappa/cli.py", line 482, in dispatch_command
    self.load_settings(self.vargs['settings_file'])
KeyError: 'settings_file'
```

**Comment by jonathanglima** [comment]
+1 for both patches

**Comment by majackson** [comment]
@jonathanglima Unfortunately, the above commit on my fork did not solve the problem. It results in `Process exited before completing request` on every command.

I'm not really sure how to go ahead tackling this problem.

**Comment by jonathanglima** [comment]
@majackson Weird. It worked for me. I'm running Zappa stable. Could it be some issue involving something specific on your fork?

**Comment by majackson** [comment]
@jonathanglima to be clear, I get the following errors:

On Zappa 0.41.2 I get the [`KeyError: u'app_function'` error](https://github.com/Miserlou/Zappa/issues/832#issue-227301588).
On Zappa 0.41.1 I get the [psycopg2 error fixed in 0.41.2](https://github.com/Miserlou/Zappa/issues/800).
On current master, I get [the `KeyError: 'settings_file'` above](https://github.com/Miserlou/Zappa/issues/832#issuecomment-300456999).
On my [own fork](https://github.com/majackson/Zappa/commit/7997035061eea9b9dcb3669e8b10f64dd5216f8c), an attempt to fix this problem, I get the following error:
```
START RequestId: fa268bb5-372f-11e7-9eb6-f7aea330d9e4 Version: $LATEST
[DEBUG]  2017-05-12T16:28:05.786Z  fa268bb5-372f-11e7-9eb6-f7aea330d9e4  Zappa Event: {'manage': 'shell --version'}
1.11
END RequestId: fa268bb5-372f-11e7-9eb6-f7aea330d9e4
REPORT RequestId: fa268bb5-372f-11e7-9eb6-f7aea330d9e4  Duration: 403.04 ms  Billed Duration: 500 ms   Memory Size: 512 MB  Max Memory Used: 58 MB
RequestId: fa268bb5-372f-11e7-9eb6-f7aea330d9e4 Process exited before completing request
```

All of this is using Django 1.11 / Python 3.6. The command I'm testing with is `zappa manage _stage_ 'shell --version'`

**Comment by Miserlou** [comment]
Added your patch to core manually. THIS NEEDS A PROPER TEST!

I think there are lots of corner cases that aren't Python3 polished yet. Help wanted.

**Comment by majackson** [comment]
👍  I think you're right, the issue above looks like a separate py3 problem. Might be a good idea to close this issue?

**Comment by Miserlou** [comment]
Thinking this is fixed in 0.42.0.
